Diagnostics/AVTester/Test-ExchAVExclusions.ps1

```diff
@@ -65,60 +65,152 @@ $LogFile = "ExchAvExclusions.log"
-$BaseFolders.Add((Join-Path $env:ExchangeInstallPath '\TransportRoles\Data\Temp').tolower())
-$BaseFolders.Add((Join-Path $env:ExchangeInstallPath '\TransportRoles\Logs').tolower())
-$BaseFolders.Add((Join-Path $env:ExchangeInstallPath '\TransportRoles\Pickup').tolower())
-$BaseFolders.Add((Join-Path $env:ExchangeInstallPath '\TransportRoles\Replay').tolower())
-$BaseFolders.Add((Join-Path $env:ExchangeInstallPath '\UnifiedMessaging\Grammars').tolower())
-$BaseFolders.Add((Join-Path $env:ExchangeInstallPath '\UnifiedMessaging\Prompts').tolower())
-$BaseFolders.Add((Join-Path $env:ExchangeInstallPath '\UnifiedMessaging\Temp').tolower())
-$BaseFolders.Add((Join-Path $env:ExchangeInstallPath '\UnifiedMessaging\Voicemail').tolower())
-$BaseFolders.Add((Join-Path $env:ExchangeInstallPath '\Working\OleConverter').tolower())
-$BaseFolders.Add((Join-Path $env:SystemDrive '\inetpub\temp\IIS Temporary Compressed Files').tolower())
-$BaseFolders.Add((Join-Path $env:SystemRoot '\Microsoft.NET\Framework64\v4.0.30319\Temporary ASP.NET Files').tolower())
-$BaseFolders.Add((Join-Path $env:SystemRoot '\System32\Inetsrv').tolower())
-
-# Checking if it is a mailbox server to avoid errors if it is an Edge server.
 if ((Get-ExchangeServer $env:COMPUTERNAME).IsMailboxServer) {
+    if (Get-DatabaseAvailabilityGroup ) {
+        if ((Get-DatabaseAvailabilityGroup).Servers.name.Contains($env:COMPUTERNAME) ) {
+            $BaseFolders.Add((Join-Path $($env:SystemRoot) '\Cluster').tolower())
+            $dag = $null
+            $dag = Get-DatabaseAvailabilityGroup | Where-Object { $_.Servers.Name.Contains($env:COMPUTERNAME) }
+            #needs local system rigths
+            if ( $null -ne $dag ) {
+                $BaseFolders.Add($("\\" + $($dag.WitnessServer.Fqdn) + "\" + $($dag.WitnessDirectory.PathName.Split("\")[-1])).ToLower())
+            }
+        }
+    }
+    $BaseFolders.Add((Join-Path $ExchangePath '\ClientAccess\OAB').tolower())
+    $BaseFolders.Add((Join-Path $ExchangePath '\FIP-FS').tolower())
+    $BaseFolders.Add((Join-Path $ExchangePath '\GroupMetrics').tolower())
+    $BaseFolders.Add((Join-Path $ExchangePath '\Logging').tolower())
+
+    $mbxS = (Get-MailboxServer -Identity $($env:COMPUTERNAME))
+    $BaseFolders.Add($($mbxS.CalendarRepairLogPath.PathName).tolower())
+    $BaseFolders.Add($($mbxS.LogPathForManagedFolders.PathName).tolower())
+
+    $BaseFolders.Add(($((Get-PopSettings).LogFileLocation)).tolower())
+    $BaseFolders.Add(($((Get-ImapSettings).LogFileLocation)).tolower())
+
+    $BaseFolders.Add((Join-Path $ExchangePath '\Mailbox').tolower())
+
     # Add all database folder paths
     foreach ($Entry in (Get-MailboxDatabase -Server $Env:COMPUTERNAME)) {
         $BaseFolders.Add((Split-Path $Entry.EdbFilePath -Parent).tolower())
         $BaseFolders.Add(($Entry.LogFolderPath.pathname.tolower()))
     }
+
+    $fetsLogs = Get-FrontEndTransportService $($env:COMPUTERNAME) | Select-Object ConnectivityLogPath, `
```

Comment:
This script was based in the following link from the comments:
https://docs.microsoft.com/en-us/Exchange/antispam-and-antimalware/windows-antivirus-software?view=exchserver-2019
And it is only for 2016/2019.
The doc for 2013 is:
https://docs.microsoft.com/en-us/exchange/anti-virus-software-in-the-operating-system-on-exchange-servers-exchange-2013-help
and it has many differences. That was the main reason because I thought it is better to control the version to 2016/9.
Because the original is based on the 2016/2019, I think is better to keep on that way and in a new version handle all the 2013 paths.
What do you think?

Reply:
I do not test it because the documentation on the comments was for 2016/2019 but looking at 2013 doc, it should work partially but will not verify some paths. 
For example, 2013 has client access role with some specific paths that are not controlled

Reply:
If right now the script does work with 2013, then we need to keep the support for 2013. We don't want it removed. till we have the additional logic in place for 2013 as well.

Reply:
if we remove:
if ( -not ( $($serverExchangeInstallDirectory.MsiProductMajor) -eq 15 -and `
        ( $($serverExchangeInstallDirectory.MsiProductMinor) -eq 1 -or $($serverExchangeInstallDirectory.MsiProductMinor) -eq 2 ) ) ) {

it will work in the same way that before the modification with 2013, in fact I think we will include more exclusion that before because the original version does not include several paths 
I just included the control of the version to be accurate with the documentation that the script follow.

In any case, I am working in a new version that works with Exchange 2013.